components/scream/src/interface/scream_scorpio_interface.F90

```diff
@@ -0,0 +1,933 @@
+    fdata_int_1d(:)   = my_dummy_value
+    fdata_int_2d(:,1) = my_dummy_value
+    fdata_int_2d(:,2) = my_dummy_value! + 1000
+    fdata_real_1d(:)   = my_dummy_r
+    fdata_real_2d(:,1) = my_dummy_r
+    fdata_real_2d(:,2) = my_dummy_r! + 1000
+    ! Record TIME
+    call get_var(pio_atm_file,'time',var)
+    ierr = pio_put_var(pio_atm_file%pioFileDesc,var%piovar,(/ pio_atm_file%numRecs /), (/ 1 /), (/ time /))
+    ! Record the outputs
+    call grid_write_data_array("example_pio_structured.nc", fdata_int_1d,  "foo")
+    call grid_write_data_array("example_pio_structured.nc", fdata_real_1d, "real_foo")
+    call grid_write_data_array("example_pio_structured.nc", fdata_real_1d, "x")
+    call grid_write_data_array("example_pio_structured.nc", fdata_real_1d, "y")
+    call grid_write_data_array("example_pio_structured.nc", fdata_real_1d, "bar2")
+    call grid_write_data_array("example_pio_structured.nc", fdata_real_2d, "bar")
+
+    call PIO_syncfile(pio_atm_file%pioFileDesc)
+
+    call get_pio_atm_file("example_pio_structured_v2.nc",pio_atm_file)
+    pio_atm_file%numRecs = pio_atm_file%numRecs + 1
+    time = pio_atm_file%numRecs * 300.0_rtype
+    my_dummy_r = 100.0_rtype*(pio_myrank+1) + pio_atm_file%numRecs
+    do ivar = 1,3
+      do jj = 1,10
+      fdata_real_1d(jj) = jj
+      fdata_real_2d(jj,ivar) = 10*ivar + jj!my_dummy_r + 1000*ivar
+      fdata_real_2d_inv(ivar,jj) = 10*ivar + jj!my_dummy_r + 1000*ivar
+      end do
+    end do
+    ! Record TIME
+    call get_var(pio_atm_file,'time',var)
+    ierr = pio_put_var(pio_atm_file%pioFileDesc,var%piovar,(/ pio_atm_file%numRecs /), (/ 1 /), (/ time /))
+    ! Record outputs
+    if (pio_atm_file%numRecs == 1) then
+      call grid_write_data_array("example_pio_structured_v2.nc", fdata_real_1d, "x")
+      call grid_write_data_array("example_pio_structured_v2.nc", fdata_real_1d, "y")
+    end if
+    call grid_write_data_array("example_pio_structured_v2.nc", fdata_real_2d_inv, "bar")
+    call grid_write_data_array("example_pio_structured_v2.nc", fdata_real_2d, "foo")
+    call grid_write_data_array("example_pio_structured_v2.nc", fdata_real_2d_inv, "bar_flip")
+    call grid_write_data_array("example_pio_structured_v2.nc", fdata_real_2d, "foo_flip")
+    call PIO_syncfile(pio_atm_file%pioFileDesc)
+    
+  end subroutine eam_history_write
+!=====================================================================!
+  subroutine eam_history_finalize()
+
+  ! For now nothing to do, could be used in the future.  If not, should be
+  ! deleted.
+
+  end subroutine eam_history_finalize
+!=====================================================================!
+  subroutine eam_pio_createHeader(File)
+
+    type(file_desc_t), intent(in) :: File             ! Pio file Handle
+    integer :: retval
+
+    ! TODO change options below to match specific simulation case
+    retval=pio_put_att (File, PIO_GLOBAL, 'source', 'SCREAM')
+    retval=pio_put_att (File, PIO_GLOBAL, 'case', 'TEST 1')
```

Comment:
How will all these info be retrieved, when I/O is complete? Is the scorpio interface module going to store them? If so, might as well do it in this PR. Are they going to be passed to the routine? If so, might as well mod the routine in this PR.

If you only need some info to test your PR, you can just use the first line, and maybe the git hash.

Edit: we should be storing our git sha in `scream_config.f`, but it seems we are not. We should add a line 
`#cmakedefine SCREAM_GIT_COMMIT_SHA "${SCREAM_GIT_COMMIT_SHA}"` in it, and have a small cmake bit that retrieve the git sha in our main CMakeLists.txt:

```
# Get sha of last scream commit
EXECUTE_PROCESS(
  COMMAND git rev-parse --abbrev-ref HEAD
  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
  OUTPUT_VARIABLE SCREAM_GIT_COMMIT_SHA
  OUTPUT_STRIP_TRAILING_WHITESPACE
)
```

Reply:
Actually, we _do_ get the sha of the last git commit, [here](https://github.com/E3SM-Project/scream/blob/91e3e9aec83ae317403094d9987ab5867f7996c7/components/scream/CMakeLists.txt#L41). However, we should a) get it also in non-CIME builds, and b) stuff the result in the `scream_config.f.in` and `scream_config.h.in` files, so we can use/print it in our runs.

Reply:
i wanted to keep track of the typical header data locally instead of going to look for it again when I was ready to start writing full headers.  The end goal is to have this information populated by the field-manager in the AD which will be in charge of the PIO files.  Once this is set up my hope is to remove `create_header` from the Fortran altogether and have the AD call these `put_att` commands directly.